src/components/inputs/rich-text-input/editor.js

```diff
@@ -0,0 +1,352 @@
+                          />
+                        }
+                      />
+                    </Tooltip>
+                    <Tooltip title="Redo" placement="bottom" off={!hasRedos}>
+                      <Button
+                        active={false}
+                        label={'redo'}
+                        isDisabled={!hasRedos}
+                        onMouseDown={editor.toggleRedo}
+                        icon={
+                          <RedoIcon
+                            color={!hasRedos ? 'neutral60' : 'solid'}
+                            size="medium"
+                          />
+                        }
+                      />
+                    </Tooltip>
+                  </ToolbarRightControls>
+                </Toolbar>
+                <div style={containerStyles}>
+                  <div ref={registerContentNode}>
+                    <EditorContainer {...props} ref={ref}>
+                      {props.children}
+                    </EditorContainer>
+                  </div>
+                </div>
+              </Container>
+              {renderToggleButton && (
+                <div
+                  css={css`
+                    display: flex;
+                    justify-content: flex-end;
+                  `}
+                >
+                  <FlatButton
+                    onClick={toggle}
+                    label={
+                      isOpen
+                        ? intl.formatMessage(messages.collapse)
+                        : intl.formatMessage(messages.expand)
+                    }
+                    icon={
+                      isOpen ? (
+                        <AngleUpIcon size="small" />
+                      ) : (
+                        <AngleDownIcon size="small" />
+                      )
+                    }
+                  />
+                </div>
+              )}
+            </Spacings.Stack>
+          </Constraints.Horizontal>
+        );
+      }}
+    </CollapsibleMotion>
+  );
+};
+
+const renderEditor = (props, editor, next) => {
+  const children = next();
+
+  const passedProps = {
+    name: props.name,
+    id: props.id,
+    isDisabled: props.disabled,
+    horizontalConstraint: props.options.horizontalConstraint,
+    defaultExpandMultilineText: props.options.defaultExpandMultilineText,
+    hasError: props.options.hasError,
+    hasWarning: props.options.hasWarning,
+    isReadOnly: props.readOnly,
+    ...filterDataAttributes(props),
+  };
+
+  return (
+    <Editor editor={editor} {...passedProps}>
+      {children}
+    </Editor>
+  );
+};
+
+const sharedProps = {
+  id: PropTypes.string,
+  name: PropTypes.string,
+  placeholder: PropTypes.string,
+  disabled: PropTypes.bool,
+  readOnly: PropTypes.bool,
+  horizontalConstraint: PropTypes.oneOf(['m', 'l', 'xl', 'scale']),
+};
+
+Editor.displayName = 'Editor';
+Editor.propTypes = { ...sharedProps, editor: PropTypes.any };
+
+renderEditor.displayName = 'renderEditor';
```

Comment:
Is this supposed to be a react component? Why not name it `RenderEditor`? 🤔

Reply:
because it's not really a react component. Check out the docs [here](https://docs.slatejs.org/guides/rendering)

Reply:
...then maybe an eslint-ignore is more adequate than adding a `displayName`?